app/Transcripts.tsx

```diff
@@ -0,0 +1,161 @@
+'use client'
+
+import { useEffect, useState } from 'react';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import Link from 'next/link';
+import { TailSpin } from 'react-loader-spinner';
+import resolveConfig from 'tailwindcss/resolveConfig';
+import { content, theme } from '../tailwind.config.cjs';
+import { VideoData, getAllVideosForDateRange } from '../utilities/metadata-utils';
+import { formatDateForPath, getVideoPath, parseDateFromPath } from 'utilities/path-utils';
+import TranscriptFilter, { TranscriptFilterSelection, DateRange } from 'components/TranscriptFilter';
+import { isValid } from 'date-fns';
+
+export default function Transcripts({ allCategories }: { allCategories: string[] }) {
+  const { filterParams, updateFilterParams } = useFilterParams(allCategories);
+  const [filters, updateFilters]: [TranscriptFilterSelection, any] = useState(filterParams);
+
+  const [isLoading, setLoading] = useState(true);
+  const [videos, setVideos]: [VideoData[], any] = useState([]);
+
+  useEffect(() => {
+    if (filters.category === null || (filters.dateRange.start === null && filters.dateRange.end === null)) {
+      setVideos([]);
+      return () => {};
+    }
+
+    let ignore = false;
+
+    const startDateString = filters.dateRange.start !== null ? formatDateForPath(filters.dateRange.start) : null;
+    const endDateString = filters.dateRange.end !== null ? formatDateForPath(filters.dateRange.end) : null;
+
+    getAllVideosForDateRange(filters.category, startDateString, endDateString)
```

Comment:
I've been trying to use await in all the code unless there's a veeeery specific reason to use the chaining syntax. Move to that?

Reply:
You can't await in a client component

Reply:
Here's some context for that: https://react.dev/reference/react/useEffect#fetching-data-with-effects
Basically, a client component function can't be async, and you can't pass an async function into useEffect. You can build and then call an async function from within the useEffect callback, but it's a lot of boilerplate for not a lot of value IMO.

One thing they recommend that I do think I want to do is move the fetch into a hook of its own, which seems a lot tidier.

Reply:
I did end up using the temporary async function when I moved the fetch into a hook, but I don't think it's any cleaner.